tests/components/broadlink/test_config_flow.py

```diff
@@ -0,0 +1,748 @@
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_flow_user_network_unreachable(hass):
+    """Test we handle a network unreachable in the user step."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", side_effect=OSError(errno.ENETUNREACH, None)):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "192.168.1.32"},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_flow_user_os_error(hass):
+    """Test we handle an OS error in the user step."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", side_effect=OSError()):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "192.168.1.32"},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "unknown"}
+
+
+async def test_flow_auth_authentication_error(hass):
+    """Test we handle an authentication error in the auth step."""
+    device = get_device("Living Room")
+    mock_api = device.get_mock_api()
+    mock_api.auth.side_effect = blke.AuthenticationError()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "reset"
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_flow_auth_device_offline(hass):
+    """Test we handle a device offline in the auth step."""
+    device = get_device("Living Room")
+    mock_api = device.get_mock_api()
+    mock_api.auth.side_effect = blke.DeviceOfflineError()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "auth"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)